src/components/games/sprint/question-list-creator/index.ts

```diff
@@ -0,0 +1,102 @@
+import WordsAPI from '../../../../api/words-api';
+import {
+  BOOK_SECTIONS,
+  BASE_URL,
+  STORAGE_KEYS,
+  DEFAULT_PAGE_NAME,
+  MAX_PAGES_IN_BOOK_SECTION,
+} from '../../../../constants';
+import { IAggregatedWord, ISprintQuestionInfo, IWord, Numbers, PageName } from '../../../../types';
+import Randomizer from '../../../../utils/randomizer';
+import RequestProcessor from '../../../request-processor';
+
+export default class QuestionListCreator {
+  private api: WordsAPI;
+
+  private randomizer: Randomizer;
+
+  private requestProcessor: RequestProcessor;
+
+  constructor() {
+    this.api = new WordsAPI();
+    this.randomizer = new Randomizer();
+    this.requestProcessor = new RequestProcessor();
+  }
+
+  public async getQuestionList(level: number, levelPage?: number): Promise<ISprintQuestionInfo[]> {
+    const currentPage =
+      (localStorage.getItem(STORAGE_KEYS.currentPage) as PageName) || DEFAULT_PAGE_NAME;
+    const result: ISprintQuestionInfo[] = [];
+
+    if (currentPage === 'studentBook') {
+      if (level === BOOK_SECTIONS.difficultWords.group) {
+        const wordList = this.randomizer.shuffle<IAggregatedWord>(
+          await this.requestProcessor.process<IAggregatedWord[]>(this.api.getDifficultWords)
+        );
+        return this.randomizer.shuffle<ISprintQuestionInfo>(this.createMinQuestionInfo(wordList));
+      }
+      for (
+        let iteration = levelPage as number;
+        iteration >= Numbers.One;
+        iteration -= Numbers.One
+      ) {
+        // eslint-disable-next-line no-await-in-loop
+        await this.addPageQuestionListToArray(result, level, iteration, levelPage);
+      }
+      return result;
+    }
+    for (
+      let iteration = Numbers.One;
+      iteration <= MAX_PAGES_IN_BOOK_SECTION;
+      iteration += Numbers.One
+    ) {
+      // eslint-disable-next-line no-await-in-loop
+      await this.addPageQuestionListToArray(result, level, iteration);
+    }
+    return this.randomizer.shuffle<ISprintQuestionInfo>(result);
+  }
+
+  private async addPageQuestionListToArray(
+    resultingArray: ISprintQuestionInfo[],
+    section: number,
+    iteration: number,
+    page?: number
+  ): Promise<void> {
+    const wordList = this.randomizer.shuffle<IWord>(await this.api.getWords(section, iteration));
+    let questionsInfo: ISprintQuestionInfo[];
+    if (page) {
```

Comment:
For student book (when page is passed) I use here additional shuffle, since the resulting array with all previous pages is not shuffled (so that first come words from current page, then previous and so on).
For start from menu (when page is not passed) here I don't use shuffle, since then the resulting array with all words from section is shuffled.